gen/definitions/generic/device.yaml

```diff
@@ -0,0 +1,64 @@
+---
+name: Device
+rest_endpoint: /device
+doc_category: Inventory
+res_description: NOT USED - please reference the data source for device list
+remove_id: True
+skip_templates: ["resource.go", "resource_test.go", "resource.tf", "import.sh"]
+attributes:
+  - model_name: board-serial
+    tf_name: serial_number
+    type: String
+    description: Serial number for device. Could be board or virtual identifier
+    example: 2AJC9DJ
+    query_param: True
+  - model_name: host-name
+    tf_name: name
+    type: String
+    query_param: true
+    description: TThe hostname of a device
+    example: va-001
+  - model_name: data
+    tf_name: devices
+    type: List
+    mandatory: true

```

Comment:
If it is only a data source, `mandatory` is not relevant, so I think it can be removed here and for any other attributes.